govcd/adminvdc.go

```diff
@@ -52,33 +95,241 @@ func (adminVdc *AdminVdc) Refresh() error {
+// Returns a Vdc pointer and an error
+func createVdc(adminOrg *AdminOrg, vdcConfiguration *types.VdcConfiguration) (*Vdc, error) {
+	util.Logger.Printf("[TRACE] createVdc called %#v", *vdcConfiguration)
+	err := adminOrg.CreateVdcWait(vdcConfiguration)
+	if err != nil {
+		return nil, err
+	}
+
+	vdc, err := adminOrg.GetVDCByName(vdcConfiguration.Name, true)
+	if err != nil {
+		return nil, err
+	}
+	return vdc, nil
+}
+
+// updateVdcAsync updates a VDC with the given params. Returns a Task and error
+func updateVdcAsync(adminVdc *AdminVdc) (Task, error) {
+	util.Logger.Printf("[TRACE] updateVdcAsync called %#v", *adminVdc)
+	adminVdc.AdminVdc.Xmlns = types.XMLNamespaceVCloud
+
+	// Return the task
+	return adminVdc.client.ExecuteTaskRequest(adminVdc.AdminVdc.HREF, http.MethodPut,
+		types.MimeAdminVDC, "error updating VDC: %s", adminVdc.AdminVdc)
+}
+
+// updateVdc updates a VDC with the given params. Returns an AdminVdc.
+func updateVdc(adminVdc *AdminVdc) (*AdminVdc, error) {
+	util.Logger.Printf("[TRACE] updateVdc called %#v", *adminVdc)
+	task, err := updateVdcAsync(adminVdc)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
 
 	err = adminVdc.Refresh()
 	if err != nil {
-		return AdminVdc{}, err
+		return nil, err
+	}
+
+	return adminVdc, nil
+}
+
+// updateVdcAsyncV97 updates a VDC with the given params. Supports Flex type allocation.
+// Needs vCD 9.7+ to work. Returns a Task and an error.
+func updateVdcAsyncV97(adminVdc *AdminVdc) (Task, error) {
+	util.Logger.Printf("[TRACE] updateVdcAsyncV97 called %#v", *adminVdc)
+	adminVdc.AdminVdc.Xmlns = types.XMLNamespaceVCloud
+
+	// Return the task
+	return adminVdc.client.ExecuteTaskRequestWithApiVersion(adminVdc.AdminVdc.HREF, http.MethodPut,
+		types.MimeAdminVDC, "error updating VDC: %s", adminVdc.AdminVdc,
+		adminVdc.client.GetSpecificApiVersionOnCondition(">= 32.0", "32.0"))
+}
+
+// updateVdcV97 updates a VDC with the given params
+// and waits for the asynchronous task to complete. Supports Flex type allocation.
+// Needs vCD 9.7+ to work. Returns an AdminVdc pointer and an error.
+func updateVdcV97(adminVdc *AdminVdc) (*AdminVdc, error) {
+	util.Logger.Printf("[TRACE] updateVdcV97 called %#v", *adminVdc)
+	task, err := updateVdcAsyncV97(adminVdc)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
+	err = adminVdc.Refresh()
+	if err != nil {
+		return nil, err
 	}
+	return adminVdc, nil
+}
+
+// createVdcAsync creates a VDC with the given params under the given organization.
+// Returns a Task and an error
+func createVdcAsync(adminOrg *AdminOrg, vdcConfiguration *types.VdcConfiguration) (Task, error) {
+	util.Logger.Printf("[TRACE] createVdcAsync called %#v", *vdcConfiguration)
+	return adminOrg.CreateVdc(vdcConfiguration)
+}
 
-	return *adminVdc, nil
+// createVdcAsyncV97 creates a VDC with the given params under the given organization
```

Comment:
Function name in the comment doesn't match actual function name.

Reply:
fixed